DqtApi/src/DqtApi/Services/LinkTrnToIdentityUserService.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using DqtApi.DataStore.Crm;
+using DqtApi.DataStore.Crm.Models;
+using DqtApi.DataStore.Sql;
+using DqtApi.Services.IdentityApi;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace DqtApi.Services
+{
+    public class LinkTrnToIdentityUserService : BackgroundService
+    {
+        private static readonly TimeSpan _pollInterval = TimeSpan.FromMinutes(1);
+        private readonly IDataverseAdapter _dataverse;
+        private readonly ILogger<LinkTrnToIdentityUserService> _logger;
+        private readonly IIdentityApiClient _identityApi;
+        private readonly IServiceProvider _serviceProvider;
+
+        public LinkTrnToIdentityUserService(IDataverseAdapter dataverse, ILogger<LinkTrnToIdentityUserService> logger, IServiceProvider serviceProvider, IIdentityApiClient client)
+        {
+            _dataverse = dataverse;
+            _logger = logger;
+            _identityApi = client;
+            _serviceProvider = serviceProvider;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            var timer = new PeriodicTimer(_pollInterval);
+            do
+            {
+                try
+                {
+                    using (var scope = _serviceProvider.CreateScope())
+                    {
+                        var dqtContext = scope.ServiceProvider.GetRequiredService<DqtContext>();
+                        var trnsNotLinkedToIndentities = dqtContext.TrnRequests.Where(x => x.LinkedToIdentity == false).ToList();
+                        foreach (var trn in trnsNotLinkedToIndentities)
+                        {
+                            var teacher = await _dataverse.GetTeacher(
+                                trn.TeacherId.Value,
+                                columnNames: new[]
+                                {
+                                    Contact.Fields.dfeta_TRN
+                                });
+
+                            try
+                            {
+                                //call api to link account to trn
+                                if (trn.IdentityUserId.HasValue && teacher is not null)
```

Comment:
Could the `trn.IdentityUserId.HasValue` part of this move into the `trnsNotLinkedToIndentities` query?

Reply:
Could put the `teacher is not null` outside the `try` and write a warning message to log if `teacher is null`